app/models/on_the_fly_calc.rb

```diff
@@ -7,163 +7,180 @@ def compute_total_score(scores)
-          end
-          @respective_scores = {}
-          @respective_scores = @review_scores[response_map.reviewer_id][round] if !@review_scores[response_map.reviewer_id].nil? && !@review_scores[response_map.reviewer_id][round].nil?
-
-          if !@corresponding_response.empty?
-            # @corresponding_response is an array, Answer.get_total_score calculate the score for the last one
-            @this_review_score_raw = Answer.get_total_score(response: @corresponding_response, questions: @questions)
-            if @this_review_score_raw
-              @this_review_score = ((@this_review_score_raw * 100) / 100.0).round if @this_review_score_raw >= 0.0
-            end
-          else
-            @this_review_score = -1.0
-          end
-
-          @respective_scores[response_map.reviewee_id] = @this_review_score
-          @review_scores[response_map.reviewer_id] = {} if @review_scores[response_map.reviewer_id].nil?
-          @review_scores[response_map.reviewer_id][round] = {} if @review_scores[response_map.reviewer_id][round].nil?
-          @review_scores[response_map.reviewer_id][round] = @respective_scores
-        end
-      end
-    else # [reviewer_id][reviewee_id] = score
+    if self.varying_rubrics_by_round?
       @response_maps = ResponseMap.where(['reviewed_object_id = ? && type = ?', self.id, @response_type])
-      review_questionnaire_id = review_questionnaire_id()
-
-      @questions = Question.where(['questionnaire_id = ?', review_questionnaire_id])
-
-      @response_maps.each do |response_map|
-        # Check if response is there
-        @corresponding_response = Response.where(['map_id = ?', response_map.id])
-        @respective_scores = {}
-        @respective_scores = @review_scores[response_map.reviewer_id] unless @review_scores[response_map.reviewer_id].nil?
-
-        if !@corresponding_response.empty?
-          # @corresponding_response is an array, Answer.get_total_score calculate the score for the last one
-          @this_review_score_raw = Answer.get_total_score(response: @corresponding_response, questions: @questions)
-          if @this_review_score_raw
-            @this_review_score = ((@this_review_score_raw * 100) / 100.0).round if @this_review_score_raw >= 0.0
-          end
-        else
-          @this_review_score = -1.0
-        end
-        @respective_scores[response_map.reviewee_id] = @this_review_score
-        @review_scores[response_map.reviewer_id] = @respective_scores
-      end
-
+      scores_varying_rubrics
+    else
+      @response_maps = ResponseMap.where(['reviewed_object_id = ? && type = ?', self.id, @response_type])
+      scores_non_varying_rubrics
     end
     @review_scores
   end
 
   # calculate the avg score and score range for each reviewee(team), only for peer-review
   def compute_avg_and_ranges_hash
     scores = {}
+    contributor_score = scores[contributor.id]
```

Comment:
Error here: undefined local variable or method `contributor' for #<Assignment:0x007f306fd0ca90>